runtime/src/in_mem_accounts_index.rs

```diff
@@ -115,7 +130,20 @@ impl<T: IndexValue> InMemAccountsIndex<T> {
         };
         Self::update_time_stat(time, m);
         Self::update_stat(count, 1);
-        result
+
+        if result.is_some() {
+            return result;
+        }
+
+        // not in cache, look on disk
+        let new_entry = self.load_account_entry_from_disk(pubkey)?;
+        let mut map = self.map().write().unwrap();
+        let entry = map.entry(*pubkey);
+        let result = match entry {
+            Entry::Occupied(occupied) => Arc::clone(occupied.get()),
+            Entry::Vacant(vacant) => Arc::clone(vacant.insert(new_entry)),
+        };
+        Some(result)
```

Comment:
Minor, can the `Arc::clone()` get refactored out of each match arm and put in the return value instead? i.e. `Some(Arc::clone(result))`.

Reply:
No, that was my first attempt, but the return is a ref in both cases and the lifetimes didn't work.